lib/postgrex/utils.ex

```diff
@@ -47,15 +47,23 @@ defmodule Postgrex.Utils do
   Fills in the given `opts` with default options.
   """
   @spec default_opts(Keyword.t) :: Keyword.t
-  def default_opts(opts) do
+  def default_opts(opts, environment \\ System) do
     opts
-    |> Keyword.put_new(:username, System.get_env("PGUSER") || System.get_env("USER"))
-    |> Keyword.put_new(:password, System.get_env("PGPASSWORD"))
-    |> Keyword.put_new(:hostname, System.get_env("PGHOST") || "localhost")
-    |> Keyword.put_new(:port, System.get_env("PGPORT"))
+    |> Keyword.put_new(:username, environment.get_env("PGUSER") || environment.get_env("USER"))
+    |> Keyword.put_new(:password, environment.get_env("PGPASSWORD"))
+    |> Keyword.put_new(:hostname, environment.get_env("PGHOST") || "localhost")
+    |> Keyword.put_new(:port, set_port(environment.get_env("PGPORT")))
```

Comment:
I can't change the environment variables because it will break users which
actually use these variables (like me).

On Mon, Feb 22, 2016 at 2:34 PM Eric Meadows-Jönsson <
notifications@github.com> wrote:

> In lib/postgrex/utils.ex
> <https://github.com/ericmj/postgrex/pull/171#discussion_r53660702>:
>
> >      opts
> > -    |> Keyword.put_new(:username, System.get_env("PGUSER") || System.get_env("USER"))
> > -    |> Keyword.put_new(:password, System.get_env("PGPASSWORD"))
> > -    |> Keyword.put_new(:hostname, System.get_env("PGHOST") || "localhost")
> > -    |> Keyword.put_new(:port, System.get_env("PGPORT"))
> > +    |> Keyword.put_new(:username, environment.get_env("PGUSER") || environment.get_env("USER"))
> > +    |> Keyword.put_new(:password, environment.get_env("PGPASSWORD"))
> > +    |> Keyword.put_new(:hostname, environment.get_env("PGHOST") || "localhost")
> > +    |> Keyword.put_new(:port, set_port(environment.get_env("PGPORT")))
>
> To make sure it always runs you can do:
>
> test "foo" do
>   ...after
>   System.delete_env(...)end
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ericmj/postgrex/pull/171/files#r53660702>.
>